### DISCUSSION TITLE
Changing the name of an ASP.NET Core application through config

### DISCUSSION THREAD

**Discussion by kimbell** [dotnet/aspnetcore#29855] [Q&A] [answered]
Looking at the [documentation ](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/host/generic-host?view=aspnetcore-5.0)for the generic host, one of the things that can be changed is the name of the application. i.e. we want a different name than the entry assembly name. Since ASP.NET Core is now based on the generic host, the assumption is that it would be possible to also change the name there. 

I extended my code with
`hostBuilder.ConfigureHostConfiguration(builder => builder.AddJsonFile("hostsettings.json", optional: true));`
```json
{
  "applicationName": "NotEntryAssembly"
}
```
In my project, I'm using the OpenApi server component to generate Swagger. This infrastructure launches the app, and it crashed and burned. 

> System.IO.FileNotFoundException: Could not load file or assembly 'NotEntryAssembly, Culture=neutral, PublicKeyToken=null'. The system cannot find the file specified.
> at System.Reflection.RuntimeAssembly.InternalLoad(ObjectHandleOnStack assemblyName, ObjectHandleOnStack requestingAssembly, StackCrawlMarkHandle stackMark, Boolean throwOnFileNotFound, ObjectHandleOnStack assemblyLoadContext, ObjectHandleOnStack retAssembly)
> at System.Reflection.RuntimeAssembly.InternalLoad(AssemblyName assemblyName, RuntimeAssembly requestingAssembly, StackCrawlMark& stackMark, Boolean throwOnFileNotFound, AssemblyLoadContext assemblyLoadContext)
> at System.Reflection.Assembly.Load(AssemblyName assemblyRef)
> at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartManager.GetApplicationPartAssemblies(String entryAssemblyName)
> at Microsoft.AspNetCore.Mvc.ApplicationParts.ApplicationPartManager.PopulateDefaultParts(String entryAssemblyName)

Digging into the sourcecode
https://github.com/dotnet/aspnetcore/blob/c925f99cddac0df90ed0bc4a07ecda6b054a0b02/src/Mvc/Mvc.Core/src/DependencyInjection/MvcCoreServiceCollectionExtensions.cs#L81-L89

this code assumes that the application name is the same as the entry assembly name.

Is there any way of changing the application name? 

Our code base is for a team that is part of a larger organization. The namespaces and assembly names are prefixed with company information. We want to use the application name within our team without the additional information.

**Comment by davidfowl** [comment, marked as answer]
I'd suggest using another interface (like making your own) as what we have today is unfortunately tied to the assembly name.
